### PR TITLE
Add debug build for GCC 5 and Clang 6

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ matrix:
     env:
     - COMPILER=g++
     - IMG=gcc-5
-    - CMAKE_EXTRA_FLAGS="-DENABLE_CUDA=Off -DENABLE_COVERAGE=On -DENABLE_C=On"
+    - CMAKE_EXTRA_FLAGS="-DENABLE_CUDA=Off -DCMAKE_BUILD_TYPE=Debug -DENABLE_COVERAGE=On -DENABLE_C=On"
```
Comment (mcfadden8) [inline .travis.yml]: Can we get Debug builds for gcc8 as well?  We also might want to run with ReleaseWithDebug?

Comment (davidbeckingsale) [reply .travis.yml]: Good suggestion. I don't want to blow up the build matrix too much though.

Comment (mcfadden8) [reply .travis.yml]: Okay.  I'll take whatever I can get..

**Comment by davidbeckingsale** [comment]
This should be ready to go @mcfadden8 @jdahm

**Pull request merged**